bqn-mode.el

```diff
@@ -519,47 +519,56 @@ If it doesn't exist, create and return it; else, return the existing one."
       (bqn-comint-mode)
       (set-input-method "BQN-Z"))))
 
-(defun bqn-comint-process-execute-region (start end &optional dont-follow)
+(defun bqn-comint--escape (str)
+  "Targeting CBQN."
+  (concat
+   ")escaped \""
+   (with-temp-buffer
+     (insert str)
+     (goto-char (point-min))
+     (while (search-forward-regexp "[\\\"\r\n]" nil 'noerror)
+       (let ((m (match-string 0)))
+         (cond
+          ((string= m "\\") (replace-match "\\\\" t t))
+          ((string= m "\"") (replace-match "\\\"" t t))
+          ((string= m (char-to-string ?\n)) (replace-match "\\n" t t))
+          ((string= m (char-to-string ?\r)) (replace-match "\\r" t t)))))
+     (buffer-string))
+   "\""))
+
+(defun bqn-comint-process-execute-region (start end &optional follow)
   "Send the region bounded by START and END to the bqn-comint-process-session.
 
-When DONT-FOLLOW is non-nil, maintain focus on the buffer where
-the function was called from."
+When FOLLOW is non-nil, switch to the inferior process buffer."
   (interactive "r")
   (when (= start end)
-    (error
-     (concat "Attempt to send empty region to "
-             bqn-comint-*process-buffer-name*)))
+    (error "Attempt to send empty region to %s" bqn-comint--process-name))
   (when bqn-comint-flash-on-send
     (bqn-comint--flash-region start end))
   (let ((region (buffer-substring-no-properties start end))
-        (session (bqn-comint-process-ensure-session))
-        (buffer (current-buffer)))
-    (pop-to-buffer (process-buffer session))
-    (goto-char (point-max))
-    (insert (format "\n%s\n" region))
-    (comint-send-input)
-    (when (or dont-follow nil)
-      (pop-to-buffer buffer))))
-
-(defun bqn-comint-process-execute-line-and-follow ()
-  "Send the current line to BQN process and focus BQN process buffer."
-  (interactive)
-  (bqn-comint-process-execute-region (line-beginning-position) (line-end-position)))
-
-(defun bqn-comint-process-execute-buffer-and-follow ()
-  "Send the current buffer to BQN process and focus BQN process buffer."
-  (interactive)
-  (bqn-comint-process-execute-region (point-min) (point-max)))
-
-(defun bqn-comint-process-execute-line ()
-  "Send the line containing the point to the BQN process."
-  (interactive)
-  (bqn-comint-process-execute-region (line-beginning-position) (line-end-position) t))
-
-(defun bqn-comint-process-execute-buffer ()
+        (pbuf (process-buffer (bqn-comint-process-ensure-session))))
+    (with-current-buffer pbuf
+      ;; get rid of prompt for output alignment
+      (goto-char (point-max))
+      (comint-kill-whole-line 0))
+    (comint-send-string (get-buffer-process pbuf)
+                        (concat (bqn-comint--escape region) "\n"))
+    (when follow
+      (select-window (display-buffer pbuf)))))
+
+(defun bqn-comint-process-execute-line (&optional arg)
+  "Send the active region, else the current line to the BQN process."
+  (interactive "P")
+  (cond
```

Comment:
Isn't `if` better than `cond` here? There is only one condition being tested, namely `(use-region-p)`.

Reply:
Done